### ISSUE TITLE
Discussion: style guide, action suffixes

### ISSUE THREAD

**Issue by zygimantas** [ngxs/store#370]
We can name things as we want, but there is a [very useful style guide](https://ngxs.gitbook.io/ngxs/recipes/style-guide) which helps the growing community to have a common language.

I have noticed that there is a recommendation NOT to add a suffix to action name. A valid action names is:
```FeedAnimal``` and we should dispatch the action like this: ```dispatch(new FeedAnimal());```

1. I am not native English speaker, but I feel a smell in ```new FeedAnimal``` part.
2. I like, and my IDE strongly pushed me to keep the methods to be alphabetically sorted, so my state action handlers are in this order: ```BuyTicket, FeedAnimal, SellTicket```

Maybe there are reasons I can't see at this moment, but why not to name action like this: ```AnimalFeedAction```? That would solve the issues above:

1. Calling ```dispatch(new AnimalFeedAction())``` feels more natural (you are dispatching an Action, not an Animal)
2. Action handlers, separate files on the disk or classes in a state file, are in this order: ```AnimalFeedAction, TicketBuyAction, TicketSellAction```. If we choose to use separate files for action classes (tslint: single class per file), the files are: ```animal-feed.action.ts, ticket-buy.action.ts, ticket-sell.action.ts```
3. Action handler in state file may have ```onAnimalFeedAction``` or even shorter ```onAnimalFeed``` name.

What do you think?

**Comment by maktarsis** [comment]
As for me, `new FeedAnimal()` is enough, but in something you are right.

Probably `FeedAnimalAction` is just OK

But perhaps, for more sense, you can do like that:
`const feedAnimalAction = new FeedAnimal();`
`dispatch(feedAnimalAction);`

**Comment by roydanton** [comment]
I prefer adding the suffix as well, with the same reasoning as zygimantas.

Because I do put all actions related to a domain in a single file and I include those actions using
`import * as domainActions from './domain.actions';` there is no need to name the actions using the object-verb pattern. I use verb-object, i.e. `FeedAnimalAction`, which is exactly how the corresponding side effect method in the service would be called.

And additionally, I like to separate the 'reducer' state from the 'side effect' state. They are two different classes, where the 'side effect' class is derived from the 'reducer' class.

**Comment by amcdnl** [comment]
Personally `FeedAnimalAction` is a bit wordy for me. I prefer shorter. I think its perfectly fine if you want to do that though.

Closing this for now.
